### PR TITLE
feat(hub): adding `downloadFileToCacheDir`

### PR DESCRIPTION
**Pull request by axel7083**
## Description

Following https://github.com/huggingface/huggingface.js/pull/1031 which added a `pathsInfo` method which can return the etag/commitHash for a given file. Allowing to be compliant with the `_hf_hub_download_to_cache_dir`[^1] method from the python library.

[^1]: [huggingface_hub/file_download.py#L882](https://github.com/huggingface/huggingface_hub/blob/c547c839dbbe0163e3ca422d017daad7c7f9361f/src/huggingface_hub/file_download.py#L882)

## Potential issue

The JS implementation do not handle the .lock files as the python library does.. This could be a problem if using the JS and PY function.. ?

The JS could make a basic implementation of the lock file that the PY lib is doing if this is a hard requirement.

## Testing

I wrote tests for the existing `downloadFile` function (no change to the implementation) and the new one added `downloadFileToCacheDir`.
 
- [x] unit tests has been added

### CODE REVIEW

File: packages/hub/src/lib/download-file.ts
[the remark this answers is not in the document]
Comment (axel7083) [reply packages/hub/src/lib/download-file.ts:1]: Good idea! (I created a new file `download-file-cache.ts`) if you have a better idea for the same, I am open to the suggestion !

[the remark this answers is not in the document]
Comment (coyotte508) [reply packages/hub/src/lib/download-file.ts:1]: If the function is `downloadFileToCacheDir`, the name of the file would be `download-file-to-cache-dir.ts`, at least that's the pattern we've been using so far

File: packages/hub/src/lib/cache-management.ts [hunk 1]
```diff
@@ -16,12 +16,19 @@ function getHuggingFaceHubCache(): string {
 	return process.env["HUGGINGFACE_HUB_CACHE"] ?? getDefaultCachePath();
 }
 
-function getHFHubCache(): string {
+export function getHFHubCache(): string {
```
Comment (coyotte508) [inline packages/hub/src/lib/cache-management.ts:19]: If it's exported, let's rename to

```suggestion
export function getHFHubLocalCachePath(): string {
```

File: packages/hub/src/lib/download-file.ts
[the remark this answers is not in the document]
Comment (coyotte508) [reply packages/hub/src/lib/download-file.ts:1]: Actually, maybe name teh function `downloadFileToLocalCache`? (and same for the file)

[the remark this answers is not in the document]
Comment (axel7083) [reply packages/hub/src/lib/download-file.ts:1]: I renamed the files :+1:

File: packages/hub/src/lib/cache-management.ts (hunk 1 above)
Comment (axel7083) [reply packages/hub/src/lib/cache-management.ts]: Renamed, you are right, this is more explicit and easier to understand.

**Comment by coyotte508** [comment]
Thanks a lot, probably a `browser: { "./download-file-to-cache-dir.ts": false}` to add

**Comment by axel7083** [comment]
> Thanks a lot, probably a `browser: { "./download-file-to-cache-dir.ts": false}` to add in `package.json`

Should be good now :) !

**Comment by coyotte508** [comment]
Can you find a way to skip the test in browser environment? (I can look into it otherwise)

File: packages/hub/vitest-browser.config.mts [hunk 2]
```diff
@@ -2,6 +2,6 @@ import { configDefaults, defineConfig } from "vitest/config";
 
 export default defineConfig({
 	test: {
-		exclude: [...configDefaults.exclude, "src/utils/FileBlob.spec.ts", "src/lib/cache-management.spec.ts"],
+		exclude: [...configDefaults.exclude, "src/utils/FileBlob.spec.ts", "src/lib/cache-management.spec.ts", "download-file-to-cache-dir.spec.ts"],
```
Comment (coyotte508) [inline packages/hub/vitest-browser.config.mts:5]:
```suggestion
		exclude: [...configDefaults.exclude, "src/utils/FileBlob.spec.ts", "src/lib/cache-management.spec.ts", "src/lib/download-file-to-cache-dir.spec.ts"],
```

Comment (axel7083) [reply packages/hub/vitest-browser.config.mts]: sorry :(

**Comment by coyotte508** [comment]
Thanks!!

**Pull request merged**